client/allocrunner/taskrunner/logmon_hook.go

```diff
@@ -142,14 +149,27 @@ func (h *logmonHook) Prestart(ctx context.Context,
 	}
 }
 
-func (h *logmonHook) isLoggingDisabled() bool {
+func (h *logmonHook) isLoggingEnabled() bool {
+	if !h.config.enabled {
+		h.logger.Debug("log rotation is disabled by task")
+		return false
+	}
+
+	// internal plugins have access to a capbility to disable logging and
+	// metrics via a private interface; this is an "experimental" interface and
+	// currently only the docker driver exposes this to users.
 	ic, ok := h.runner.driver.(drivers.InternalCapabilitiesDriver)
 	if !ok {
-		return false
+		return true
 	}
 
 	caps := ic.InternalCapabilities()
-	return caps.DisableLogCollection
+	if caps.DisableLogCollection {
+		h.logger.Debug("log rotation is disabled by driver")
```

Comment:
Is log rotation right here or does this relate to log collection? I can't figure out how this would look to an operator going through logs.